go/common/crypto/tls/cert.go

```diff
@@ -52,13 +52,7 @@ func LoadOrGenerate(certPath, keyPath, commonName string) (*tls.Certificate, err
 	return cert, nil
 }
 
-// Generate generates a new TLS certificate.
-func Generate(commonName string) (*tls.Certificate, error) {
-	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
-	if err != nil {
-		return nil, fmt.Errorf("tls: failed to generate keypair: %w", err)
-	}
-
+func makeCertificate(commonName string, pubKey, privKey interface{}) (*tls.Certificate, error) {
```

Comment:
Should not take `interface{}` for the keys, we require Ed25519 keys during certificate verification, so would be good to make it type safe.